### PR TITLE
Adding node16 image

### PR DESCRIPTION
**Pull request by alrod**

### CODE REVIEW

File: host/4/publish-appservice-stage.yml [hunk 1]
```diff
@@ -96,6 +96,34 @@ steps:
...
+      docker tag $SOURCE_REGISTRY/node:$(PrivateVersion)-node16-appservice $TARGET_REGISTRY/node:$(TargetVersion)-node16-appservice-stage$(StageNumber)
+
+      docker push $TARGET_REGISTRY/node:$(TargetVersion)-appservice-stage$(StageNumber)
+      docker push $TARGET_REGISTRY/node:$(TargetVersion)-node16-appservice-stage$(StageNumber)
+
+      docker system prune -a -f
+    displayName: tag and push node images
+    continueOnError: false    
+
+  - bash: |
```
Comment (balag0) [inline host/4/publish-appservice-stage.yml]: Is this the same as above section?

File: host/4/bullseye/amd64/node/node16/node16-appservice.Dockerfile [hunk 2]
```diff
@@ -0,0 +1,95 @@
+# Build the runtime from source
+ARG HOST_VERSION=4.0.0-preview.5.16714
+FROM mcr.microsoft.com/dotnet/sdk:6.0.100-rc.2 AS runtime-image
+ARG HOST_VERSION
+
+# Build requires 3.1 SDK
+COPY --from=mcr.microsoft.com/dotnet/sdk:5.0 /usr/share/dotnet /usr/share/dotnet
```
Comment (pragnagopa) [inline host/4/bullseye/amd64/node/node16/node16-appservice.Dockerfile]: this is probably not needed. .Net 5 SDK for needed to build V3 runtime

**Review by pragnagopa — CHANGES_REQUESTED** [review]
no need to install : .Net 5 SDK as this was only needed to build V3

**Comment by pragnagopa** [comment]
@CooperLink  - please do take a look - more images coming in !

File: host/4/bullseye/amd64/node/node16/node16.Dockerfile [hunk 3]
```diff
@@ -0,0 +1,72 @@
...
+
+RUN EXTENSION_BUNDLE_VERSION=1.8.1 && \
+    EXTENSION_BUNDLE_FILENAME=Microsoft.Azure.Functions.ExtensionBundle.1.8.1_linux-x64.zip && \
+    apt-get update && \
+    apt-get install -y gnupg wget unzip && \
+    wget https://functionscdn.azureedge.net/public/ExtensionBundles/Microsoft.Azure.Functions.ExtensionBundle/$EXTENSION_BUNDLE_VERSION/$EXTENSION_BUNDLE_FILENAME && \
+    mkdir -p /FuncExtensionBundles/Microsoft.Azure.Functions.ExtensionBundle/$EXTENSION_BUNDLE_VERSION && \
+    unzip /$EXTENSION_BUNDLE_FILENAME -d /FuncExtensionBundles/Microsoft.Azure.Functions.ExtensionBundle/$EXTENSION_BUNDLE_VERSION && \
+    rm -f /$EXTENSION_BUNDLE_FILENAME && \
+    EXTENSION_BUNDLE_VERSION_V2=2.6.1 && \
```
Comment (pragnagopa) [inline host/4/bullseye/amd64/node/node16/node16.Dockerfile]: please update to latest extension bundles . See https://github.com/Azure/azure-functions-docker/commit/9e6ddfd52a39f554886a808e4cddb9c3771d0f93 for reference

**Review by pragnagopa — APPROVED** [review]
@balag0 - these changes need to part of V4 GA payload as well. Please take a look

File: host/4/bullseye/amd64/node/node16/node16-appservice.Dockerfile [hunk 4]
```diff
@@ -0,0 +1,92 @@
...
+
+RUN EXTENSION_BUNDLE_VERSION=1.8.1 && \
+    EXTENSION_BUNDLE_FILENAME=Microsoft.Azure.Functions.ExtensionBundle.1.8.1_linux-x64.zip && \
+    apt-get update && \
+    apt-get install -y gnupg wget unzip && \
+    wget https://functionscdn.azureedge.net/public/ExtensionBundles/Microsoft.Azure.Functions.ExtensionBundle/$EXTENSION_BUNDLE_VERSION/$EXTENSION_BUNDLE_FILENAME && \
+    mkdir -p /FuncExtensionBundles/Microsoft.Azure.Functions.ExtensionBundle/$EXTENSION_BUNDLE_VERSION && \
+    unzip /$EXTENSION_BUNDLE_FILENAME -d /FuncExtensionBundles/Microsoft.Azure.Functions.ExtensionBundle/$EXTENSION_BUNDLE_VERSION && \
+    rm -f /$EXTENSION_BUNDLE_FILENAME && \
+    EXTENSION_BUNDLE_VERSION_V2=2.6.1 && \
```
Comment (balag0) [inline host/4/bullseye/amd64/node/node16/node16-appservice.Dockerfile]: looks like this version has an issue, please update to 2.8.3 (and in other files as well if required)
https://github.com/Azure/azure-functions-docker/commit/9e6ddfd52a39f554886a808e4cddb9c3771d0f93

**Comment by balag0** [comment]
> @balag0 - these changes need to part of V4 GA payload as well. Please take a look

@alrod 
Some minor comments. Once merged please work with @kshyju on building the image.

**Pull request merged**